### PR TITLE
Feature/recipe class

### PR DESCRIPTION
**Pull request by FrazierLE**
This branch includes:
Unit tests for recipe class & recipeRepository class
Code for recipe class & recipeRepository class
Worked: Eleanor & Lauren
WIP unit tests for recipeRepository class, specifically the createRecipe method
Left comments in for createRecipe method test and lines 24-26 of recipeRepository class for testing purposes of the real data
Commented out lines 1 & 2 of recipe class -- still unsure if we need it or not for future iterations/user stories
Is there a more efficient way to do this?
Indicate anything that still needs to be addressed

### CODE REVIEW

File: src/classes/Recipe.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            this.ingredients.forEach((item) => {
+                if (item.id === ingredient.id)
+                  {acc.push(ingredient.name)};
+              });
+              return this.ingredientsList = acc;
+          }, []);
+            return this.ingredientsList;
+    }
+    calculateCost(ingredientInfo) {
+        this.totalCost = ingredientInfo.reduce((acc, ingredient) => {
```
Comment (knpausch) [inline src/classes/Recipe.js:26]: *optional: rename acc to total

File: test/Recipe-test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+describe('Recipe', () => {
+  sampleIngredientsData
+  sampleUsersData
+  sampleRecipeData
+  let recipe
+
+
+  beforeEach(() => {
+    recipe = new Recipe(sampleRecipeData[0]);
```
Comment (knpausch) [inline test/Recipe-test.js:13]: You may pass a whole object in the argument of Recipe so we can see the data. Example: recipe = new Recipe({id: 20081, name: wheat flour, estimatedCostInCents: 142})

File: test/Recipe-test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    sampleIngredientsData;
+    sampleUsersData;
+    sampleRecipeData;
+  });
+
+  it('Should be a function', () => {
+    expect(Recipe).to.be.a('function');
+  });
+  it('should hold property of id', () => {
+    expect(recipe.id).to.equal(sampleRecipeData[0].id)
```
Comment (knpausch) [inline test/Recipe-test.js:23]: hardcode to.equals to a visible data value. Example: expect(recipe.id).to.equal(20081). Repeat this for lines 26, 29, 32,  35, and 38.

File: test/Recipe-test.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    expect(recipe.tags).to.deep.equal(sampleRecipeData[0].tags)
+  });
+  it("should hold property of ingredient list", () => {
+    expect(recipe.ingredientsList).to.deep.equal([])
+  });
+  it("determine the names of ingredients needed", () => {
+    expect(recipe.determineIngredients(sampleIngredientsData)).to.deep.equal(['wheat flour', 'bicarbonate of soda', 'eggs', 'sucrose'])
+  });
+  it("should hold property of total cost", () => {
+    expect(recipe.totalCost).to.equal()
```
Comment (knpausch) [inline test/Recipe-test.js:47]: there is not value set for this .to.equal

**Review by knpausch — APPROVED** [review]
Great tests and utilizing iterators in the recipe class! Comments are assigned to their line numbers for your consideration. Unit testing is great. To improve it, you may set their value to a hardcode value (see comments for example).

**Review by jmyacobn — APPROVED** [review]
Kiko and I reviewed together, so please the comments in her review. Overall, the tests look great, but need to incorporate the hard coded values from the sample data rather than dynamic values.

File: src/classes/Recipe.js (hunk 1 above)
Comment (Eleanorgruth) [reply src/classes/Recipe.js:26]: We changed that! Thank you

File: test/Recipe-test.js (hunk 4 above)
Comment (Eleanorgruth) [reply test/Recipe-test.js:47]: We set this to 'undefined'. This property isn't assigned until the calculateCost() method is invoked.

**Pull request merged**